### PR TITLE
Use Async Inference Config when available for endpoint update

### PR DESCRIPTION
**Pull request by shreyapandit**
*Issue #, if available:*

*Description of changes:*

Endpoint update fails because Async Config is not present 
>   File "src/sagemaker_scripts/deploy_endpoint.py", line 110, in <module>
    accelerator_type=args.accelerator_instance,
  File "/local/home/andrdodd/Sources/eeyore-amber-cdk/src/EeyoreML/.venv/lib64/python3.7/site-packages/sagemaker/predictor.py", line 307, in update_endpoint
    self.endpoint_name, new_endpoint_config_name, wait=wait
  File "/local/home/andrdodd/Sources/eeyore-amber-cdk/src/EeyoreML/.venv/lib64/python3.7/site-packages/sagemaker/session.py", line 3084, in update_endpoint
    EndpointName=endpoint_name, EndpointConfigName=endpoint_config_name
  File "/local/home/andrdodd/Sources/eeyore-amber-cdk/src/EeyoreML/.venv/lib64/python3.7/site-packages/botocore/client.py", line 391, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/local/home/andrdodd/Sources/eeyore-amber-cdk/src/EeyoreML/.venv/lib64/python3.7/site-packages/botocore/client.py", line 719, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the UpdateEndpoint operation: The AsyncInferenceConfig parameter cannot be removed from this endpoint hescatterplot-2022-05-19-08-06-08-445. To change the endpoint

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/session.py [hunk 1]
```diff
@@ -3055,6 +3055,11 @@ def create_endpoint_config_from_existing(
         if request_data_capture_config_dict is not None:
             request["DataCaptureConfig"] = request_data_capture_config_dict
 
+        if existing_endpoint_config_desc.get("AsyncInferenceConfig") is not None:
```
Comment (navinsoni) [inline src/sagemaker/session.py:3058]: can you please change " existing_endpoint_config_desc.get("AsyncInferenceConfig") " to " existing_endpoint_config_desc.get("AsyncInferenceConfig", None) "

Comment (shreyapandit) [reply src/sagemaker/session.py:3058]: Sure

Comment (navinsoni) [reply src/sagemaker/session.py:3058]: Oh I see it now I meant it on line 58

**Review by navinsoni — APPROVED** [review]

**Review by mufaddal-rohawala — APPROVED** [review]

**Review by navinsoni — APPROVED** [review]
/bot run all

**Review by navinsoni — COMMENTED** [review]
/bot run all

**Pull request merged**